api/src/test/java/org/asynchttpclient/test/TestUtils.java

```diff
@@ -92,6 +99,57 @@ public static File createTempFile(int approxSize) throws IOException {
         }
     }
 
+    public static class SyncByteBufferPublisher implements Publisher<ByteBuffer> {
+        // FIXME If I reduce the chunk size the test will fail. Why?
+        private static final int CHUNK_SIZE = 30000;
+
+        private final byte[] bytes; // payload streamed to the subscriber
+        private final int chunkSize; // number of bytes in each element sent to
+                                     // the subscriber
+
+        public SyncByteBufferPublisher(byte[] bytes) {
+            this(bytes, CHUNK_SIZE);
+        }
+
+        public SyncByteBufferPublisher(byte[] bytes, int chunkSize) {
+            this.bytes = Arrays.copyOf(bytes, bytes.length);
+            this.chunkSize = chunkSize;
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super ByteBuffer> subscriber) {
+            subscriber.onSubscribe(new Subscription() {
+                // This index is used to keep track of bytes that have been sent.
+                private volatile int currentIndex = 0;
+                // This flag will track whether this `Subscription` is to be considered cancelled or not
+                private volatile boolean cancelled = false;
+
+                public void request(long n) {
```

Comment:
If we were to go with Akka, I guess we would have to turn the maven project into a Scala/Java mixed one so we can use TestKit. I'm afraid this would make the project more difficult to build for most people. So, even thought my heart would love to go with Akka, I think it's more reasonable to stay with a simple Java project and go with rx. WDYT?